akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -280,33 +280,28 @@ private[stream] object Collect {
 
       import Collect.NotApplied
 
-      var recovered: Option[T] = None
+      var recovered: OptionVal[T] = OptionVal.none
 
-      override def onPush(): Unit = {
-        push(out, grab(in))
-      }
+      override def onPush(): Unit = push(out, grab(in))
```

Comment:
we typically prefer a newline here, but not important
(the `{` is indeed superfluous)